### PR TITLE
Talljoe's layout with common keymap.

### PR DESCRIPTION
**Pull request by Talljoe**
This does a fair amount of fun to create a common keymap that works across a variety of keyboards! Since it has some larger-scale changes (i.e. renaming a layout to a better name) submitting a PR now before I work on 40% support.

### CODE REVIEW

File: users/talljoe/rules.mk [hunk 1]
```diff
@@ -0,0 +1,7 @@
+SRC += talljoe.c
+
+ifneq ("$(wildcard users/$(KEYMAP)/config.h)","")
```
Comment (drashna) [inline users/talljoe/rules.mk]: This isn't needed. It's part of the main build script now: https://github.com/qmk/qmk_firmware/blob/master/build_keyboard.mk#L219-L221

File: build_keyboard.mk [hunk 2]
```diff
@@ -204,7 +204,9 @@ else
 endif
 
 # User space stuff
-USER_PATH := users/$(KEYMAP)
+ifeq ("$(USER_PATH)","")
```
Comment (drashna) [inline build_keyboard.mk]: (@jackhumbert) is this needed?

Comment (drashna) [reply build_keyboard.mk]: Ah, nice!

Though, if you're adding this, it should be added to the documentation too, so that others know they can do this.

File: users/talljoe/talljoe.h [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+#endif
+#define LY_QWER   DF(_BASE)
+#define LY_WORK   DF(_WORKMAN)
+#define LY_NRMN   DF(_NORMAN)
+#define LY_DVRK   DF(_DVORAK)
+#define LY_CLMK   DF(_COLMAK)
+#define TG_NKRO   MAGIC_TOGGLE_NKRO
+#define KC_PTT    KC_F24
+#define MS_MID    KC_MS_BTN3
+#define FX(x)     (EFFECT + x)
```
Comment (drashna) [inline users/talljoe/talljoe.h]: Neat! :)

File: build_keyboard.mk (hunk 2 above)
Comment (Talljoe) [reply build_keyboard.mk]: Done, and made it a bit more user friendly.

File: autocomplete.sh [hunk 4]
```diff
@@ -1 +1 @@
...
...dale preonic-default preonic-default_ortho_5x12 preonic-jacwib preonic-kinesis preonic-nikchi preonic-smt preonic-xyverz preonic-zach ps2avrGB-default roadkit-default roadkit-default_ortho_4x4 roadkit-flipphone roadkit-mjt roadkit-singles roadkit-singlesBrent s60_x-ansi_qwertz s60_x-bluebear s60_x-custom s60_x-dbroqua s60_x-default s60_x-hasu s60_x-hhkb s60_x-iso s60_x-jpec s60_x-plain s60_x-poker s60_x-poker_bit s60_x-poker_set s60_x-spacefn s65_x-default s65_x-iso s65_x-nall s65_x-smt satan-admiralStrokers satan-ben_iso satan-colemak satan-dbroqua satan-default satan-default_60_ansi satan-default_60_hhkb satan-default_60_iso satan-denolfe satan-iso_split_rshift satan-mark1 satan-midi satan-poker satan-rask63 satan-sethbc satan-smt satan-stanleylai satan-talljoe satan-unxmaal sixkeyboard-default subatomic-default sweet16-default tada68-default tada68-maartenwut tada68-rgb tiger_lily-default tv44-belak tv44-core tv44-default tv44-jeebak tv44-jetpacktuxedo tv44-mjt tv44-smt tv44-tong92 tv44-xyverz uk78-default vision_division-default whitefox-default whitefox-jetpacktuxedo whitefox-matt3o xd60-cheese xd60-default xd60-stanleylai xd75-default xd75-default_ortho_5x15 xd75-fabian" make
```
Comment (drashna) [inline autocomplete.sh]: Did you edit this, or was this a auto-formatting change?

Comment (Talljoe) [reply autocomplete.sh]: Search and Replace. ansi_split_bs_rshift -> hhkb

**Comment by drashna** [comment]
This looks good to go, then. 

However, there is a pretty big change (the userspace thing, which 👍 personally), but also the split shift -> HHKB and the added HHKB layout.    Which I think is a good idea, as well. 

I'd rather let @jackhumbert or @skullydazed take a look and verify that it's okay.

**Comment by Talljoe** [comment]
I can back out the change. I thought it was a bit wordy and it turns out there's a standard-ish name for it. I also see that someone has copied it for iso (`iso_split_bs_rshift`) so I'm fine without the rename.

**Comment by Talljoe** [comment]
Backed out layout name change.

**Comment by drashna** [comment]
In that case...

Thanks!
